src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -37,13 +38,34 @@ void setUserStore(UserStore userStore) {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request,response);
+        //Checks whether user is logged in or not
+        String userName = (String) request.getSession().getAttribute("user");
```

Comment:
Lets do some error checking on this userName string. See ChatServlet#doPost

Reply:
By error checking you mean verifying whether user is logged in or not.

Reply:
+1

Reply:
Nope. Verify that the string is not null

Reply:
Fiixed it!